services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/SubnetPortsRepository.java

```diff
@@ -55,35 +58,10 @@ public SubnetPortsRepository(CacheFactory cacheFactory) {
 
             for (PortEntity.FixedIp fixedIp: fixedIps) {
                 String subnetId = fixedIp.getSubnetId();
-                if (!subnetPortIdsMap.containsKey(subnetId)) {
-                    SubnetPortIds subnetPortIds = new SubnetPortIds(subnetId, new HashSet<>());
-                    subnetPortIdsMap.put(subnetId, subnetPortIds);
-                }
-
-                subnetPortIdsMap.get(subnetId).getPortIds().add(portEntity.getId());
-            }
-        }
-
-        return new ArrayList<>(subnetPortIdsMap.values());
-    }
-
-
-    public void addSubnetPortIds(List<PortEntity> portEntities) throws Exception {
-        //Store the mapping between subnet id and port id
-        List<SubnetPortIds> subnetPortIdsList = getSubnetPortIds(portEntities);
-
-        for (SubnetPortIds item: subnetPortIdsList) {
-            String subnetId = item.getSubnetId();
-            Set<String> portIds = item.getPortIds();
-
-            SubnetPortIds subnetPortIds = subnetPortIdsCache.get(subnetId);
-            if (subnetPortIds == null) {
-                subnetPortIds = new SubnetPortIds(subnetId, new HashSet<>(portIds));
-            } else {
-                subnetPortIds.getPortIds().addAll(portIds);
+                CacheConfiguration cfg = CommonUtil.getCacheConfiguration(subnetId);
+                ICache<String, String> cache = cacheFactory.getCache(String.class, cfg);
+                cache.put(portEntity.getId(), "1");
```

Comment:
@DavidLiu506 Can you put some comments here to describe why you add these codes here and why you put "1" in the cache?

Reply:
After discussion with @DavidLiu506, it seems that the dynamic cache creation is pretty costly so we would like to revert to previous implementation but with other improvement.

Reply:
@DavidLiu506 Please update the PR once the new change is ready.